src/de/jost_net/JVerein/gui/action/LehrgangsartDeleteAction.java

```diff
@@ -51,7 +54,16 @@ public void handleAction(Object context) throws ApplicationException
       {
         return;
       }
-
+      DBIterator<Lehrgang> it = Einstellungen.getDBService()
+          .createList(Lehrgang.class);
+      it.addFilter("lehrgangsart = ?", new Object[] { l.getID() });
+      it.setLimit(1);
+      if (it.hasNext())
+      {
+        throw new ApplicationException(String.format(
+            "Lehrgangsart '%s' kann nicht gel�scht werden. Es existiern Lehrg�nge dieser Art.",
+            l.getBezeichnung()));
+      }
```

Comment:
Da sollte sicherheitshalber der Foreign Key in Lehrgang von CASCADE nach RESTRICT geändert werden.

Reply:
Auch im extra pr